operator/pkg/model/translation/envoy_virtual_host.go

```diff
@@ -320,6 +320,7 @@ func requestMirrorMutation(mirrors []*model.HTTPRequestMirror) routeActionMutati
 func timeoutMutation(backend *time.Duration, request *time.Duration) routeActionMutation {
 	return func(route *envoy_config_route_v3.Route_Route) *envoy_config_route_v3.Route_Route {
 		if backend == nil && request == nil {
+			route.Route.Timeout = durationpb.New(0)
```

Comment:
i think it would be enough to set the `max_stream_duration` instead of the `timout` in case both backend and request timeout are empty. This way we should behave the same as with Cilium in version 1.14 (except if Gateway API with specific timeouts is used)

```
route.Route.MaxStreamDuration = &envoy_config_route_v3.RouteAction_MaxStreamDuration{
	MaxStreamDuration: &duration.Duration{
		Seconds: 0,
	},
}

```